dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/util/ValidationUtils.java

```diff
@@ -44,10 +44,7 @@ public boolean isDateInRange(OffsetDateTime timestamp) {
 		if (timestamp.isAfter(Instant.now().atOffset(ZoneOffset.UTC))) {
 			return false;
 		}
-		if (timestamp.isBefore(Instant.now().atOffset(ZoneOffset.UTC).minus(retentionPeriod))) {
-			return false;
-		}
-		return true;
+		return !timestamp.isBefore(Instant.now().atOffset(ZoneOffset.UTC).minus(retentionPeriod));
```

Comment:
Personally I like the old style better. I think it is better readable, since a) the ! is easily overlooked and b) humans are not good with negation.

Reply:
OK - I was following IntelliJ's complaint that the `if` can be rewritten as a `return` only. What happens if the `!isBefore` is changed to `isAfter`? Do we lose an exact match?

As this is called with a `timestamp` that is set to `atStartOfDay`, there could be a refusal around midnight with a clock-drift on the client. Is it OK to be ignored here?